src/main/java/Bob.java

```diff
@@ -0,0 +1,207 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Bob {
+    private static final String SEPARATOR = "____________________________________________________________";
+    private static final String LINE_PREFIX = "    ";
+
+    private static String argument = "";
+    private static final List<Task> list = new ArrayList<>();
+
+    private enum Command {
+        BYE("bye") {
+            @Override
+            public void run() {
+                exit();
+            }
+        },
+        LIST("list") {
+            @Override
+            public void run() {
+                if (list.isEmpty()) {
+                    say("You have not added any tasks yet.");
+                } else {
+                    StringBuilder text = new StringBuilder();
+                    int i;
+                    for (i = 0; i < list.size() - 1; ++i) {
+                        text.append(i + 1).append(".").append(list.get(i)).append("\n");
```

Comment:
Is there anyway that this could be abstracted? For example, maybe a method inside the Task class or its inheriting classes?